### PR TITLE
docs: `response contains property` guide

### PR DESCRIPTION
**Pull request by bandantonio**
## What/Why/How?

This PR adds an updated guide on how to use a new built-in `response-contains-property` rule based on common examples

Deprecates #602 

## Reference

https://redocly.com/docs/cli/guides/response-contains-property/

## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: docs/guides/response-contains-property.md [hunk 1]
```diff
@@ -3,231 +3,337 @@ tocMaxDepth: 3
...
-};
-
-module.exports = {
-  id,
-  rules,
-};
+If you're using the Redocly OpenAPI extension for VS Code, this is how the lint errors look
+in the integrated terminal (Problems tab):
+
+![](./assets/response-contains-property-range.png)
```
Comment (adamaltman) [inline docs/guides/response-contains-property.md:220]:
```suggestion
![response contains property problem](./images/response-contains-property-range.png)
```
issue: Link was broken, and always include an alt-text description for accessibility.

File: docs/guides/response-contains-property.md [hunk 2]
```diff
@@ -3,231 +3,337 @@ tocMaxDepth: 3
...
+
+❌ Validation failed with 3 errors.
+run `openapi lint --generate-ignore-file` to add all problems to the ignore file.
 ```
 
-### Report the problem
+If you're using the Redocly OpenAPI extension for VS Code, this is how the lint errors look
+in the integrated terminal (Problems tab):
+
+![](./assets/response-contains-property-one.png)
```
Comment (adamaltman) [inline docs/guides/response-contains-property.md:132]:
```suggestion
![Response contains property problem in VS Code](./images/response-contains-property-one.png)
```

File: docs/guides/response-contains-property.md [hunk 3]
```diff
@@ -3,231 +3,337 @@ tocMaxDepth: 3
...
+❌ Validation failed with 7 errors.
+run `openapi lint --generate-ignore-file` to add all problems to the ignore file.
 ```
 
-Now that you've followed along in the creation of a rule, you can try to craft your own.
-What if all requests should have a particular header parameter?
+If you're using the Redocly OpenAPI extension for VS Code, this is how the lint errors look
+in the integrated terminal (Problems tab):
+
+![](./images/response-contains-property-multiple.png)
```
Comment (adamaltman) [inline docs/guides/response-contains-property.md:328]:
```suggestion
![Response contains property problem in VS Code](./images/response-contains-property-multiple.png)
```

File: docs/guides/response-contains-property.md [hunk 4]
```diff
@@ -3,231 +3,337 @@ tocMaxDepth: 3
...
-- `_links` (or equivalent)
+This guide describes how to ensure that the API definition [contains a specific property](../resources/built-in-rules.md) and multiple properties.
 
-:::attention
+For this purpose, we will be checking the following cases:
 
-If there are exceptions to the rule, the exceptions can be explicitly added to the ignore file.
+1. Single response containing one property
+1. Multiple responses containing one property
+1. Multiple properties for multiple responses
```
Comment (adamaltman) [inline docs/guides/response-contains-property.md:17]:
```suggestion
1. Multiple responses containing multiple properties
```

(Seems inconsistent with the structure of 1 and 2)

File: docs/guides/response-contains-property.md [hunk 5]
```diff
@@ -3,231 +3,337 @@ tocMaxDepth: 3
...
 
-### Traverse for evaluation
+Once completed, in the project folder, execute the `npm test` command. You should receive confirmation that the OpenAPI definition is valid:
 
-```mermaid
-graph LR
-    Response --successful responses only--> MediaType --application/json only--> Schema
-```
+```bash
+npm test
```
[the remark this answers is not in the document]
Comment (RomanHotsiy) [reply docs/guides/response-contains-property.md:58]: +1 for redocly lint

[the remark this answers is not in the document]
Comment (bandantonio) [reply docs/guides/response-contains-property.md:58]: In the `openapi-starter` repository, `npm test` runs `redocly lint` under the hood, which is convenient as you don't have to install `redocly-cli` globally. If this is the case, we have to add a note telling that the package must be installed globally.

[the remark this answers is not in the document]
Comment (bandantonio) [reply docs/guides/response-contains-property.md:57]: @adamaltman @RomanHotsiy ^^

[the remark this answers is not in the document]
Comment (adamaltman) [reply docs/guides/response-contains-property.md:57]: I missed this. With a local package install, the user can run: `npx redocly lint`. The user does not need to install redocly-cli globally in that case.

File: docs/guides/response-contains-property.md [hunk 6]
```diff
@@ -3,231 +3,336 @@ tocMaxDepth: 3
...
+        ├── path-item.yaml
+        └── users_{username}.yaml
+```
+
+:::note
+Make sure that the @redocly/cli has version `1.0.0-beta.99` or later
 :::
 
-## Steps to writing the rule
+- If you're using VS Code as your favorite code editor, we recommend you to install the [Redocly OpenAPI VS Code extension](../../redocly-openapi/index.md).
```
Comment (adamaltman) [inline docs/guides/response-contains-property.md:44]:
```suggestion
- If you're using VS Code as your favorite code editor, we recommend you install the [Redocly OpenAPI VS Code extension](../../redocly-openapi/index.md).
```

File: docs/guides/response-contains-property.md [hunk 7]
```diff
@@ -3,231 +3,324 @@ tocMaxDepth: 3
...
-};
-
-module.exports = {
-  id,
-  rules,
-};
+If you're using the Redocly OpenAPI extension for VS Code, this is how the lint errors look
+in the integrated terminal (Problems tab):
+
+![Response contains property problem in VS Code](./assets/response-contains-property-range.png)
```
Comment (adamaltman) [inline docs/guides/response-contains-property.md:210]:
```suggestion
![Response contains property problem in VS Code](./images/response-contains-property-range.png)
```

File: docs/guides/response-contains-property.md [hunk 8]
```diff
@@ -3,231 +3,324 @@ tocMaxDepth: 3
...
+
+❌ Validation failed with 3 errors.
+run `openapi lint --generate-ignore-file` to add all problems to the ignore file.
 ```
 
-### Report the problem
+If you're using the Redocly OpenAPI extension for VS Code, this is how the lint errors look
+in the integrated terminal (Problems tab):
+
+![Response contains property problem in VS Code](./assets/response-contains-property-one.png)
```
Comment (adamaltman) [inline docs/guides/response-contains-property.md:125]:
```suggestion
![Response contains property problem in VS Code](./images/response-contains-property-one.png)
```

**Pull request merged**